### PR TITLE
Added Darkbox loading fullsized images animation

### LINKED ISSUE #95: Add loading annimation
- [x] Add a loading animation while loading the full size images
- [x] Add configuration for different loading animations
- [x] Document all available configuration options
- [ ] Update the complex example

### PR DESCRIPTION
**Pull request by hramberger**
Closes #95

### CODE REVIEW

**Comment by hramberger** [comment]
This should be merged after #68

File: README.md [hunk 1]
```diff
@@ -77,6 +78,15 @@ TBD
...
 | NONE      | There is a fixed end in both directions                         |
 
+##### `LoadingAnimation`
+| Name      | Description |
+|-----------|-------------|
+| DOT       | A line of jumping dots moving in a wave like form     |
+| BAR       | Three vertical bars getting wider and smaller         |
+| FLEX_RING | A ring flexing in and out                             |
+| [SPINNER](https://youtu.be/3V8nl8v24cQ)   | A classical spinner animation turning infinitely      |
+| SQUARE    | A line of jumping squared moving in a wave like form  |
```
Comment (Boerni1995) [inline README.md:88]:
```suggestion
| SQUARE    | A line of jumping squares moving in a wave like form  |
```

File: projects/ngx-darkbox-gallery/src/lib/model/darkbox-configuration.ts [hunk 2]
```diff
@@ -13,6 +13,16 @@ export interface DarkboxConfiguration {
...
    */
   enableCaption?: boolean;
+
+  /**
+   * Loading animation shown while the full sized image is loading
+   * DOT: A line of jumping dots moving in a wave like form
+   * BAR: Three vertical bars getting wider and smaller
+   * FLEX_RING: A ring flexing in and out
+   * SPINNER: A classical spinner animation turning infinitely
+   * SQUARE: A line of jumping squared moving in a wave like form
```
Comment (Boerni1995) [inline projects/ngx-darkbox-gallery/src/lib/model/darkbox-configuration.ts:23]:
```suggestion
   * SQUARE: A line of jumping squares moving in a wave like form
```

File: README.md [hunk 3]
```diff
@@ -64,10 +64,11 @@ export class AppComponent {
...
 #### `DarkboxConfiguration`
-| Name          | Type                              | Default               | Description |
-|---------------|-----------------------------------|-----------------------|-------------|
-| loopDirection | [LoopDirection](#LoopDirection)   | `LoopDirection.BOTH`  |Sets the direction the darkbox will allow looping through the images |
-| enableCaption | bool                              | `true`                |Sets if the caption should be displayed or not|
+| Name              | Type                              | Default               | Description |
+|-------------------|-----------------------------------|-----------------------|-------------|
+| loopDirection     | [LoopDirection](#LoopDirection)   | `LoopDirection.BOTH`  |Sets the direction the darkbox will allow looping through the images |
+| enableCaption     | bool                              | `true`                |Sets if the caption should be displayed or not|
+| loadingAnimation  | [LoadingAnimation](#LoadingAnimation) | Loading animation shown while the full sized image is loading|
```
Comment (Boerni1995) [inline README.md:71]:
```suggestion
| loadingAnimation  | [LoadingAnimation](#LoadingAnimation) | `LoadingAnimation.SPINNER` | Loading animation shown while the full sized image is loading|
```

**Review by Boerni1995 — CHANGES_REQUESTED** [review]

**Review by Boerni1995 — APPROVED** [review]

**Pull request merged**